### ISSUE TITLE
req.param(array) strange interpretation...

### ISSUE THREAD

**Issue by oogre** [balderdashy/sails#2462]
Hello,
I'm building my second app with Sails. It's a great framework.
I've a trouble with this case :
I send an ajax request on "PUT /controller/update/:id"
The data's request look like this : 
```
{
	_csrf : "<%= _csrf %>",
	task : array (this array may contain more than 20 id)
}
```
When data.task.length <= 20 it's ok . it runs
but when data.task.length > 20 it KO.

So, I made some investigations /config/routes.js :
```
'put /work/update/:id?' : function(req, res, next){
    console.log(req.param("task"));
    return next();
  },
```
When data.task.length <= 20 it's ok . req.param("task") => ["id", "id", ..., "id"]
but when data.task.length > 20 it KO.req.param("task") => {"0":"id", "1":"id", ..., "n":"id"} WTF?

where is that interpretation of my data from?
How can I fix it?

thank you

**Comment by netrevisanto** [comment]
Well, i have a problem taht it seems like this is the thread to post into.
I have an app written in sails 0.9.8 and i have sent arrays from my view to my server with ajax, like this:

var arr = [ {id: 1, text: 'bla'}, {id:2, text: 'bla'} ];

$.post("/blah, { arr: arr});

And in my controller:

var arr = req.param("arr");

console.log(arr);

Would output: 

 [ {id: 1, text: 'bla'}, {id:2, text: 'bla'} ];

Then, i migrated my app to current 0.10.5 version and my output is now:

{ '0': {id: 1, text: 'bla'} }, { '1': { id:2, text: 'bla'} }

And this breaks all my logic. What should i do?

**Comment by CWyrtzen** [comment]
@oogre @Josebaseba @netrevisanto is this still an issue?

**Comment by Josebaseba** [comment]
@CWyrtzen I think that you can close it, I don't have any problem sending arrays. And to send really long ones the solution is using Stringify method. Thanks.

**Comment by oogre** [comment]
@CWyrtzen It's ok for me. Actually, I switched on meteor now

**Comment by sailsbot** [comment]
Thanks for posting, @oogre.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by mikermcneil** [comment]
For future reference, out of curiosity, we just tested this with a PUT request and a JSON-encoded body and it's definitely fixed now:

![image](https://user-images.githubusercontent.com/618009/32377891-18e48d36-c077-11e7-95c5-ff0a2becef0a.png)
